From 24e6ad23fede5165ff555d0a773a1c39ec104072 Mon Sep 17 00:00:00 2001
From: elif dede <edede@twitter.com>
Date: Mon, 17 Nov 2014 16:53:08 -0800
Subject: [PATCH 50/93] PARQUET-135: Input location is not getting set for the getStatistics in ParquetLoader when using two different loaders within a Pig script.

Author: elif dede <edede@twitter.com>

Closes #86 from elifdd/parquetLoader_error_PARQUET-135 and squashes the following commits:

b0150ee [elif dede] fixed white space
bdb381a [elif dede] PARQUET-135: Call setInput from getStatistics in ParquetLoader to fix ReduceEstimator errors in pig jobs

Conflicts:
	parquet-hadoop/src/test/java/parquet/format/converter/TestParquetMetadataConverter.java
Resolution:
    Upstream patch 251a495 has only a whitespace change for this file,
    the conflict was in an area not backported. No change to the file.
---
 .../src/main/java/parquet/pig/ParquetLoader.java   |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/parquet-pig/src/main/java/parquet/pig/ParquetLoader.java b/parquet-pig/src/main/java/parquet/pig/ParquetLoader.java
index 91dc0da..8b92eca 100644
--- a/parquet-pig/src/main/java/parquet/pig/ParquetLoader.java
+++ b/parquet-pig/src/main/java/parquet/pig/ParquetLoader.java
@@ -287,8 +287,9 @@ public class ParquetLoader extends LoadFunc implements LoadMetadata, LoadPushDow
   public ResourceStatistics getStatistics(String location, Job job)
       throws IOException {
     if (DEBUG) LOG.debug("LoadMetadata.getStatistics(" + location + ", " + job + ")");
-    // We do not need to call setInput
-    // as setLocation is guaranteed to be called before this
+    /* We need to call setInput since setLocation is not
+       guaranteed to be called before this */
+    setInput(location, job);
     long length = 0;
     try {
       for (InputSplit split : getParquetInputFormat().getSplits(job)) {
-- 
1.7.0.4

